packages/instant/src/components/erc20_asset_amount_input.tsx

```diff
@@ -110,10 +122,19 @@ export class ERC20AssetAmountInput extends React.Component<ERC20AssetAmountInput
             currentFontSizePx: fontSizePx,
         });
     };
-    private readonly _handleSymbolClick = () => {
-        if (this.props.onSelectAssetClick) {
-            this.props.onSelectAssetClick(this.props.asset);
+    private readonly _generateSelectAssetClickHandler = (): (() => void) | undefined => {
```

Comment:
stuff like `canSelectOtherAssets && this._renderChevronIcon()` would have to be used in two places, so I still prefer the conditional rendering logic to be in `_renderChevronIcon` but did clean up the code a bit to be close to what you describe.

Reply:
I appreciate the helper function you introduced.

Right now `_renderChevronIcon` is conceptually really `_renderChevronIconIfApplicable`.  I still like my suggested psuedocode a bit better, as it lets the functions `onSelectAssetClick` and `_renderChevronIcon` assume that there are available assets, instead of having to check about it internally.

That being said, this isn't a huge deal -- up to your discretion if you want to leave or change